src/samples/GroupChannelPolls.js

```diff
@@ -0,0 +1,1119 @@
+      params.data = {"text": "Wrong answer!"};
+      params.isAnonymous = isPollAnonymous;
+      params.allowUserSuggestion = allowUserSuggestion;
+      params.allowMultipleVotes = allowMultipleVotes;
+      params.closeAt = -1;
+
+      if(pollMessageToUpdate) {
+        await currentlyJoinedChannel.updatePoll(pollMessageToUpdate._poll.id, params)
+
+        updateState({
+          ...stateRef.current,
+          pollTitleValue: "",
+          pollOptionValue: "",
+          pollOptionsArray: [],
+          isPollAnonymous: false,
+          allowUserSuggestion: false,
+          allowMultipleVotes: false,
+          isCreatePollModalOpen: false
+        });
+
+      } else {
+        const pollData = await sb.poll.create(params);
+
+        const messageParams = {
+          message: pollData.title,
+          pollId: pollData.id
+        };
+  
+        currentlyJoinedChannel.sendUserMessage(messageParams)
+          .onSucceeded((message) => {
+  
+            updateState({
+              ...stateRef.current,
+              pollTitleValue: "",
+              pollOptionValue: "",
+              pollOptionsArray: [],
+              isPollAnonymous: false,
+              allowUserSuggestion: false,
+              allowMultipleVotes: false,
+              isCreatePollModalOpen: false
+            });
+          })
+            .onFailed((error) => {
+              console.log(error)
+              console.log("failed")
+            });
+      }
+
+    }
+
+    const isShowCreatePollModal = (value) => {
+      switch(value) {
+        case "open":
+          updateState({ ...state, isCreatePollModalOpen: true});
+          break;
+        case "close":
+          updateState({ ...state, isCreatePollModalOpen: false, pollTitleValue: "", pollOptionsArray: []});
+          break;
+        default:
+          console.log("error");
+          break;
+      }
+    }
+
+    const addPollOption = () => {
+      if(state.pollOptionValue) {
+        updateState({ ...state, pollOptionsArray: [...state.pollOptionsArray, state.pollOptionValue], pollOptionValue: ""})
+      }
+    }
+
+    const removeOption = (e) => {
+      const newPollOptionsArray = state.pollOptionsArray.filter((value) => value !== e.currentTarget.innerHTML)
+      updateState({ ...state, pollOptionsArray: newPollOptionsArray })
+    }
+
+    const onPollCheckboxInputsChange = (e, name) => {
+      updateState({ ...state, [name]: !state[name] })
+    }
+
+    const addOrRemoveVoice = async(e, optionId, pollId) => {
+      const { userIdInputValue, checkedOptions } = state;
+
+      switch(e.currentTarget.type) {
+        case "checkbox":
+          if(e.currentTarget.checked) {
+            const newCheckedOptions = checkedOptions.slice(0);
+            newCheckedOptions.push(optionId);
+
+            updateState({ ...state, checkedOptions: newCheckedOptions });
+
+            await fetch(`https://api-${process.env.REACT_APP_SEND_BIRD_ID}.sendbird.com/v3/polls/${pollId}/vote`,{
```

Comment:
We want to avoid calling the platform api from the client side as that api is designed for server to server communication.

Michelle has put together a polls sample for a blog post. Its all done with client side sdk and should serve as good inspiration to get this working without calling the API. 

https://github.com/sendbird/sendbird-uikit-polls-sample